c/serverStatusService.c

```diff
@@ -0,0 +1,262 @@
+  JsonObject *logLevels = jsonObjectGetObject(context->serverConfig, "logLevels");
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  if(logLevels){
+    jsonPrintObject(out, logLevels);
+  }
+  jsonEnd(out);
+  finishResponse(response);
+}
+
+void respondWithServerEnvironment(HttpResponse *response, ServerAgentContext *context){
+  /*Information about parameters for smf_unc: https://www.ibm.com/support/knowledgecenter/SSLTBW_2.1.0/com.ibm.zos.v2r1.erbb700/smfp.htm#smfp*/
+  struct utsname unameRet;
+  char smfVarBuffer[64];
+  char hostnameBuffer[256];
+  char* buffer;
+  int rc = 0;
+  int reqtype = 0x00000005; //fullword. request type
+  int rectype = 0x0000004F; //SMF record type, only type 79 is supported
+  int subtype = 0x00000009; //SMF record subtype
+  int bufferlen = 716800; // Yes the SMF buffer is actually 700Kb roughly
+  int cpuUtil = 0x00000000;
+  int demandPaging = 0x00000000;
+  int options = 0x00000000;
+  int mvsSrm = 0x00000000;
+  int zaapUtil = 0x00000000;
+  int ziipUtil = 0x00000000;
+  uname(&unameRet);
+  gethostname(hostnameBuffer, sizeof(hostnameBuffer));
+  buffer = (char *)safeMalloc(bufferlen, "buffer");
+  memset(buffer, 0, bufferlen);
+  smfFunc = (EXSMFI *)fetch("ERBSMFI");
+  rc = (*smfFunc)(&reqtype,
+               &rectype,
+               &subtype,
+               buffer,
+               &bufferlen,
+               &cpuUtil,
+               &demandPaging,
+               &options,
+               &mvsSrm,
+               &zaapUtil,
+               &ziipUtil);
+  if(rc > 0){
+    respondWithError(response, HTTP_STATUS_BAD_REQUEST, "Unable to fetch from RMF data interface service");
+  }
+  safeFree(buffer, bufferlen);
+  jsonPrinter *out = respondWithJsonPrinter(response);
+  setResponseStatus(response, 200, "OK");
+  setDefaultJSONRESTHeaders(response);
+  writeHeader(response);
+  jsonStart(out);
+  jsonAddString(out, "logDirectory", getenv("ZSS_LOG_FILE"));
+  jsonAddString(out, "agentName", "zss");
+  jsonAddString(out, "agentVersion", context->productVersion);
+  jsonAddString(out, "arch", unameRet.sysname);
+  jsonAddString(out, "osRelease", unameRet.release);
+  jsonAddString(out, "osVersion", unameRet.version);
+  jsonAddString(out, "hardwareIdentifier", unameRet.machine);
+  jsonAddString(out, "hostname", hostnameBuffer);
+  jsonAddString(out, "nodename", unameRet.nodename);
+  jsonStartObject(out, "userEnvironment");
+  char *env_var = *environ;
+  for (int i = 1; env_var; i++) {
+    char *len = strchr(env_var, '=');
+    if(len == NULL){
+      break;
+    }
+    char *name = safeMalloc(strlen(env_var)-strlen(len) + 1, "env_var name");
+    snprintf(name, sizeof(char)*(strlen(env_var)-strlen(len) + 1), "%s", env_var);
+    jsonAddString(out, name, len+1);
+    safeFree(name, strlen(name));
+    env_var = *(environ+i);
+  }
+  jsonEndObject(out);
+  snprintf(smfVarBuffer, sizeof(smfVarBuffer), "%d%%", demandPaging);
```

Comment:
Why do we need to convert integers to strings? Client of this service now have to parse these values (and take into account the fact that some values have a percent sign).

Can we do ```jsonAddInt```?